src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -218,26 +219,30 @@ class ImageReplayerAdminSocketHook : public AdminSocketHook {
     for (Commands::const_iterator i = commands.begin(); i != commands.end();
 	 ++i) {
       (void)admin_socket->unregister_command(i->first);
-      delete i->second;
     }
     commands.clear();
   }
 
   bool call(std::string command, cmdmap_t& cmdmap, std::string format,
 	    bufferlist& out) override {
-    Mutex::Locker locker(lock);
-    Commands::const_iterator i = commands.find(command);
-    assert(i != commands.end());
+    CommandRef command_ref;
+    {
+      Mutex::Locker locker(lock);
+      Commands::const_iterator i = commands.find(command);
+      assert(i != commands.end());
+      command_ref = i->second;
+    }
     Formatter *f = Formatter::create(format);
     stringstream ss;
-    bool r = i->second->call(f, &ss);
+    bool r = command_ref->call(f, &ss);
```

Comment:
Is it possible to just eliminate the unique lock owned by this class and instead pass a reference to the ```ImageReplayer``` lock to it's constructor?